packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -160,8 +160,11 @@ mixin CupertinoRouteTransitionMixin<T> on PageRoute<T> {
 
   @override
   bool canTransitionTo(TransitionRoute<dynamic> nextRoute) {
-    // Don't perform outgoing animation if the next route is a fullscreen dialog.
-    return nextRoute is CupertinoRouteTransitionMixin && !nextRoute.fullscreenDialog;
+    // Don't perform outgoing animation if the next route is a fullscreen dialog,
+    // or there is no matching transition to use.
+    return ((nextRoute is! PageRoute<T>) || !nextRoute.fullscreenDialog) &&
+      ((nextRoute is ModalRoute<T> && nextRoute.delegatedTransition != null) ||
+      nextRoute is CupertinoRouteTransitionMixin);
```

Comment:
Nit: Would this be any clearer if it was broken up into named variables? Same for other canTransitionTo methods in this PR. Sorry if I mentioned this already, I can't remember at this point.